### PR TITLE
[INFRA-1955] Define an openvpn virtual machine and its dns record

### PR DESCRIPTION
**Pull request by olblak**
This PR describe a vpn machine seated on two subnetworks, one open to  the internet "public_app", and a second closed to the internet.
It also introduce a new dns record "vpn.jenkins.io" in order to reach its public endpoint with a more convenient name.

The purpose of this machine is to deploy an vpn in order to reach hidden service.
This machine is designed to be used with [jenkins-infra/openvpn](https://github.com/jenkins-infra/openvpn) and managed by [jenkins-infra/jenkins-infra](https://github.com/jenkins-infra/jenkins-infra/pull/1204)

### CODE REVIEW

File: scripts/init-puppet.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+curl -k https://puppet.jenkins.io:8140/packages/current/install.bash | sudo bash
```
Comment (rtyler) [inline scripts/init-puppet.sh]: just out of my own curiosity, I thought cloud_data scripts were executed as root already.

Comment (olblak) [reply scripts/init-puppet.sh]: Good point

File: plans/machine_vpn.tf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    sku       = "18.04-LTS"
+    version   = "latest"
+  }
+
+  storage_os_disk {
+    name              = "${var.prefix}vpn"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Standard_LRS"
+    disk_size_gb      = "50"
```
[the remark this answers is not in the document]
Comment (olblak) [reply plans/machine_vpn.tf]: True but my subconscient is traumatized by disk space issues

[the remark this answers is not in the document]
Comment (olblak) [reply plans/machine_vpn.tf]: :face_with_head_bandage:

File: plans/machine_vpn.tf [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+resource "azurerm_network_interface" "vpn_public_dmz" {
+  name                = "${var.prefix}-vpn-public-dmz"
+  location            = "${azurerm_resource_group.vpn.location}"
+  resource_group_name = "${azurerm_resource_group.vpn.name}"
+  enable_ip_forwarding          = true
+
+  ip_configuration {
+    name                          = "${var.prefix}-public-app"
+    subnet_id                     = "${azurerm_subnet.public_dmz.id}"
+    private_ip_address_allocation = "dynamic" # nope
```
[the remark this answers is not in the document]
Comment (olblak) [reply plans/machine_vpn.tf]: :nerd_face:

File: plans/dns_jenkinsio.tf [hunk 4]
```diff
@@ -175,3 +175,11 @@ resource "azurerm_dns_mx_record" "spamtrap_jenkinsio_mx_entries" {
     exchange   = "mxb.mailgun.org"
   }
 }
+
+resource "azurerm_dns_a_record" "vpn" {
+  name                = "${azurerm_public_ip.vpn.name}"
```
[the remark this answers is not in the document]
Comment (olblak) [reply plans/dns_jenkinsio.tf]: True this value shouldn't be computed

**Pull request merged**